src/FlagBag.php

```diff
@@ -34,7 +34,9 @@ class FlagBag
     private string $type;
 
     /**
-     * @param class-string<\BackedEnum>|\BackedEnum $enumOrType
+     * @param class-string<\BackedEnum> $enumType
+     *
+     * @throws \ReflectionException
```

Comment:
Such errors are like logic exceptions: it should lead to a fix in the code and are never meant to be caught, so it's not useful to document it in the docblocks.